handlers/analytics.go

```diff
@@ -0,0 +1,130 @@
+package handlers
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/golang/glog"
+	"github.com/julienschmidt/httprouter"
+	"github.com/livepeer/catalyst-api/errors"
+	"github.com/mmcloughlin/geohash"
+	"github.com/xeipuuv/gojsonschema"
+	"io"
+	"net/http"
+	"strconv"
+)
+
+const GEO_HASH_PRECISION = 3
+
+type AnalyticsHandler struct {
+}
+
+type AnalyticsLog struct {
+	SessionID  string              `json:"session_id"`
+	PlaybackID string              `json:"playback_id"`
+	Protocol   string              `json:"protocol"`
+	PageURL    string              `json:"page_url"`
+	SourceURL  string              `json:"source_url"`
+	Player     string              `json:"player"`
+	UserAgent  string              `json:"user_agent"`
+	UID        string              `json:"uid"`
+	Events     []AnalyticsLogEvent `json:"events"`
+}
+
+type AnalyticsLogEvent struct {
+	Type           string `json:"type"`
+	Timestamp      int64  `json:"timestamp"`
+	Errors         int    `json:"errors,omitempty"`
+	PlaytimeMS     int    `json:"playtime_ms,omitempty"`
+	TTFFMS         int    `json:"ttff_ms,omitempty"`
+	PreloadTimeMS  int    `json:"preload_time_ms,omitempty"`
+	AutoplayStatus string `json:"autoplay_status,omitempty"`
+	BufferMS       int    `json:"buffer_ms,omitempty"`
+	ErrorMessage   string `json:"error_message,omitempty"`
+}
+
+type AnalyticsGeo struct {
+	GeoHash     string
+	Continent   string
+	Country     string
+	Subdivision string
+	Timezone    string
+}
+
+type AnalyticsHandlersCollection struct {
+}
+
+func (d *AnalyticsHandlersCollection) Log() httprouter.Handle {
+	schema := inputSchemasCompiled["AnalyticsLog"]
+
+	return func(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+		log, err := parseAnalyticsLog(r, schema)
+		if log == nil {
+			errors.WriteHTTPBadRequest(w, "Invalid request payload", err)
+			return
+		}
+		geo, err := parseAnalyticsGeo(r)
+		if err != nil {
+			glog.Warning("error parsing geo info from analytics log request header, err=%v", err)
+		}
+
+		// TODO: ENG-1650, Process analytics data and remove logging
+		glog.Info("Processing analytics log: log=%v, geo=%v", log, geo)
+	}
+}
+
+func parseAnalyticsGeo(r *http.Request) (*AnalyticsGeo, error) {
```

Comment:
maybe AnalyticsGeo could just be a value rather than a pointer, it's just a simple struct and not sure there's a need for it to be a pointer?

Reply:
Updated in 966e638a1cc48551294ffc960a8f82364098cb9c